src/views/Drawer/DrawerSidebar.js

```diff
@@ -85,9 +86,21 @@ class DrawerSidebar extends PureComponent<void, Props, void> {
   _onItemPress = ({ route, focused }: DrawerItem) => {
     this.props.navigation.navigate('DrawerClose');
     if (!focused) {
-      const subAction = route.index !== undefined && route.index !== 0 // if the child screen is a StackRouter then always navigate to its first screen (see #1914)
-        ? NavigationActions.navigate({ routeName: route.routes[0].routeName })
-        : undefined;
+      let subAction = undefined;
```

Comment:
There was a bug in Flow previously that was hiding a type error here.

`route` is a `NavigationRoute`, which is a `NavigationLeafRoute | NavigationStateRoute`. A `NavigationStateRoute` is defined as a `NavigationLeafRoute`, but with a couple extra parameters. (Defs [here](https://github.com/react-community/react-navigation/blob/master/src/TypeDefinition.js#L45-L72).)

Since `NavigationLeafRoute` is not an exact type, there is no guarantee that it _doesn't_ have a `index` parameter. Thus, just because a `route` does indeed have an `index`, does not necessarily mean it is a `NavigationStateRoute` and consequently has a `routes` as well.

Either we need to use exact types in defining `NavigationRoute`, or we need to explicitly establish at run-time that `route.routes[0].routeName` exists and is a string before we treat it as such.